rtc::Network must be defined before classes that use it as an instance variable

ddkilzer opened the pull request:
#### a0c0389741bc2cee159362ee6c71d02f610f4f1c
<pre>
rtc::Network must be defined before classes that use it as an instance variable
&lt;<a href="https://bugs.webkit.org/show_bug.cgi?id=278201">https://bugs.webkit.org/show_bug.cgi?id=278201</a>&gt;
&lt;<a href="https://rdar.apple.com/133994317">rdar://133994317</a>&gt;

Reviewed by Youenn Fablet.

Fixes the following error when compiling with newer clang:

    Source/webrtc/rtc_base/network.h:192:18: note: in instantiation of member function &apos;std::unique_ptr&lt;rtc::Network&gt;::~unique_ptr&apos; requested here
      192 | class RTC_EXPORT NetworkManagerBase : public NetworkManager {
          |                  ^
    Source/webrtc/rtc_base/network.h:192:18: note: in implicit destructor for &apos;rtc::NetworkManagerBase&apos; first required here
    [...]
    In file included from Source/webrtc/p2p/base/connection.h:38:
    Source/webrtc/p2p/base/port_interface.h:31:7: note: forward declaration of &apos;rtc::Network&apos;
       31 | class Network;
          |       ^
    1 error generated.

* Source/ThirdParty/libwebrtc/Source/webrtc/rtc_base/network.h:
- Move the class definition of rtc::Network above the
  rtc::NetworkManagerBase class to fix the error.

Canonical link: <a href="https://commits.webkit.org/282603@main">https://commits.webkit.org/282603@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/bc3f06d96554c0b3846d74906c56a23809541645

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/63660 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/43016 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/16257 "Built successfully") | [  ~~🛠 wpe~~](https://ews-build.webkit.org/#/builders/5/builds/67681 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/59/builds/14268 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/50704 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/14548 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/5/builds/67681 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 wincairo-tests~~](https://ews-build.webkit.org/#/builders/59/builds/14268 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/66729 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/47/builds/39899 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/55109 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/5/builds/67681 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/36581 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/64/builds/12488 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/13141 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/58085 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/63/builds/12813 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/69377 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/7607 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/62/builds/12379 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/58562 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/7640 "Built successfully") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/8/builds/55202 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/58776 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 vision-wk2~~](https://ews-build.webkit.org/#/builders/88/builds/6327 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/9625 "Built successfully and passed tests") | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/44/builds/38837 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/46/builds/39916 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/43/builds/41028 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/45/builds/39659 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
<!--EWS-Status-Bubble-End-->

youennf reviewed (approved):
Can you add a WEBRTC_WEBKIT_BUILD for now?
I am working on having a webkit libwebrtc branch (so far https://github.com/youennf/libwebrtc/tree/webkit-m128, but hopefully it would move to the webkit repo) but for now, having WEBRTC_WEBKIT_BUILD will help keeping this change.

Commit:
rtc::Network must be defined before classes that use it as an instance variable
<https://bugs.webkit.org/show_bug.cgi?id=278201>
<rdar://133994317>

Reviewed by Youenn Fablet.

Fixes the following error when compiling with newer clang:

    Source/webrtc/rtc_base/network.h:192:18: note: in instantiation of member function 'std::unique_ptr<rtc::Network>::~unique_ptr' requested here
      192 | class RTC_EXPORT NetworkManagerBase : public NetworkManager {
          |                  ^
    Source/webrtc/rtc_base/network.h:192:18: note: in implicit destructor for 'rtc::NetworkManagerBase' first required here
    [...]
    In file included from Source/webrtc/p2p/base/connection.h:38:
    Source/webrtc/p2p/base/port_interface.h:31:7: note: forward declaration of 'rtc::Network'
       31 | class Network;
          |       ^
    1 error generated.

* Source/ThirdParty/libwebrtc/Source/webrtc/rtc_base/network.h:
- Move the class definition of rtc::Network above the
  rtc::NetworkManagerBase class to fix the error.

Canonical link: https://commits.webkit.org/282603@main
